### PR TITLE
doc entry for MultiHeadAttention

### PR DESCRIPTION
**Pull request by CarloLucibello**
References in the documentation for #2146

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
 * Fixed various deprecation warnings, from `Zygone.@nograd` and `Vararg`.
+* Initial support for `AMDGPU` via extension mechanism.
+* Add `gpu_backend` preference to select GPU backend using `LocalPreference.toml`.
+* Add `Flux.gpu_backend!` method to switch between GPU backends.
 
 ## v0.13.13
 * Added `f16` which changes precision to `Float16`, recursively.
-* Initial support for AMDGPU via extension mechanism.
-* Add `gpu_backend` preference to select GPU backend using `LocalPreference.toml`.
-* Add `Flux.gpu_backend!` method to switch between GPU backends.
```
Comment (CarloLucibello) [inline NEWS.md:11]: these were misplaced

File: docs/src/models/nnlib.md [hunk 2]
```diff
@@ -2,9 +2,20 @@
...
+```@docs
+NNlib.dot_product_attention
+NNlib.dot_product_attention_scores
+NNlib.make_causal_mask
+```
+
 ## Softmax
 
-`Flux`'s `logitcrossentropy` uses `NNlib.softmax` internally.
+`Flux`'s [`logitcrossentropy`](@ref) uses [`NNlib.logsoftmax`](@ref) internally.
```
Comment (Saransh-cpp) [inline docs/src/models/nnlib.md:18]:
```suggestion
`Flux`'s [`Flux.logitcrossentropy`](@ref) uses [`NNlib.logsoftmax`](@ref) internally.
```

File: docs/src/models/nnlib.md [hunk 3]
```diff
@@ -60,7 +72,7 @@ pixel_shuffle
 
 ## Batched Operations
 
-`Flux`'s `Bilinear` layer uses `NNlib.batched_mul` internally.
+`Flux`'s [`Bilinear`](@ref) layer uses [`NNlib.batched_mul`](@ref) internally.
```
Comment (Saransh-cpp) [inline docs/src/models/nnlib.md:75]:
```suggestion
`Flux`'s [`Flux.Bilinear`](@ref) layer uses [`NNlib.batched_mul`](@ref) internally.
```

**Comment by CarloLucibello** [comment]
ready for merge

**Review by ToucheSir — APPROVED** [review]

**Pull request merged**